lib/wanda_web/schemas/v1/catalog/check.ex

```diff
@@ -96,10 +100,32 @@ defmodule WandaWeb.Schemas.V1.Catalog.Check do
               description: "Message returned when the check fails"
             }
           },
-          required: [:name, :type, :expression]
+          required: [:name, :type, :expression, :failure_message]
         }
+      },
+      when: %Schema{
+        type: :string,
+        description: "Expression to determine whether a check should run",
+        nullable: true
+      },
+      premium: %Schema{
+        type: :boolean,
+        description: "Check is Premium or not"
       }
     },
-    required: [:id, :name, :severity, :facts, :values, :expectations]
+    required: [
+      :id,
+      :name,
+      :group,
+      :description,
+      :remediation,
+      :metadata,
+      :severity,
+      :facts,
+      :values,
+      :expectations,
+      :when,
```

Comment:
I don't know if we should put `when` here as it is nullable. Having `null` value passes the `required` test?
Anyway, not a big deal. Maybe it is good to have it in the required list

Reply:
I just wanted to have everything what is represented in the struct, having nil seems not to harm.